.github/workflows/gradle-run-ui-tests.yml

```diff
@@ -61,9 +73,27 @@ jobs:
           target: google_apis
           script: ./gradlew runAcceptanceTests
         env:
-            GITHUB_USER: ${{ github.actor }}
-            GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-            
+          GITHUB_USER: ${{ github.actor }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Publish HTML Report
+        uses: actions/upload-artifact@v4
+        with:
+          name: Acceptance Test Report
+          path: app/build/reports/androidTests/connected/debug/flavors/${{ github.event.inputs.FLAVOR }}/*.html
+
+      - name: Get commit hash
+        id: get_commit_hash
+        run: echo "::set-output name=commit_hash::$(git rev-parse HEAD)"
+
+      - name: Archive test results
+        run: |
+          zip -r integration-tests-android_${{ steps.get_commit_hash.outputs.commit_hash }}.zip app/build/reports/androidTests/connected/debug/flavors/${{ github.event.inputs.FLAVOR }}
+        uses: actions/upload-artifact@v4
+        with:
+          name: integration-tests-android_${{ steps.get_commit_hash.outputs.commit_hash }}
+          path: integration-tests-android_${{ steps.get_commit_hash.outputs.commit_hash }}.zip
```

Comment:
Awesome 👏  ! I think we forgot to have it for the `unitTests` too so I'll create a task for it, but this is what we need 🙌